platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/tipset/TipsetAdvancementWeight.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.event.tipset;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * Stores a weighed tipset advancement score.
+ * <p>
+ * If zero stake nodes were not a thing, we could use a long as a tipset advancement score. Or, if it was ok to ignore
+ * zero stake nodes, we could do the same as well. But since we don't want to allow zero sake nodes to get stale events,
+ * we need to have a mechanism for separately tracking when zero stake nodes have advanced.
```

Comment:
Let's not use the term `stake` in the platform. That is an application level concept.

Reply:
Renamed to use "weight". It's mostly just because we have been saying "stake" for so long that it has become muscle memory.